actors/builtin/miner/miner_state.go

```diff
@@ -1118,6 +1118,9 @@ func (st *State) AdvanceDeadline(store adt.Store, currEpoch abi.ChainEpoch) (*Ad
 
 	// No live sectors in this deadline, nothing to do.
 	if deadline.LiveSectors == 0 {
+		// TODO: do we still need to clear the post submissions here? I
+		// think we're technically fine, but this is a strange
+		// edge-case.
```

Comment:
I think yes, they should be cleared. 

It's a strange edge case that can be directly reached by a miner, simply by terminating a deadline's sectors. Unclear how this may interact with `ChallengeWindowedPoSt`, but I don't think we want the old snapshots to stick around!

Reply:
So, I don't think this can happen anymore (after this patch). Given this patch:

1. I can't "prove" no sectors. We reject those proofs.
2. I can't terminate sectors during the challenge window (or the previous deadline).

Given this, I shouldn't _need_ to touch this array, right?

Reply:
I think it should be done for safety, regardless. If the snapshots aren't reset each EOD cron, they may be technically valid when submitted to `ChallengeWindowedPoSt`

Reply:
I guess I can just check without spending too much time. That, and log an error.